content/reference/rest-api/templates/create.md

```diff
@@ -0,0 +1,52 @@
+---
+title: 'Templates: create'
+weight: 10
+
+menu:
+  main:
+    name: 'create'
+    identifier: 'rest-api-templates-create'
+    parent: 'rest-api-templates'
+    pre: 'Creates a new template.'
+---
+
+Creates a new template.
+
+# Request
+
+## HTTP Request
+
+```
+POST https://cawemo.com/api/v1/templates
```

Comment:
We could also offer the publishing as an optional parameter for the create and update actions. This way, users would only need to make one call to update and publish a template.

Reply:
@volkergersabeck I added a query parameter named `publishVersion` - do you think it is necessary to offer the possibility to give a name to the published version?

Reply:
I don't think sending query parameters in `POST` or `PATCH` is a good practice.
I think we should simply process the `published` boolean field from the body in both `create` and `update`, according to our resource representation: https://github.com/camunda/camunda-docs-cawemo/blob/8794-public-api/content/reference/rest-api/templates/_index.md#resource-representation.

Reply:
> I don't think sending query parameters in `POST` or `PATCH` is a good practice.

Sending query parameters in `POST` and `PATCH` requests is a widely used practice across the industry. To give you some examples:

- Twitter Rest API: https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/mute-block-report-users/api-reference/post-blocks-create
- Google Drive Rest API: https://developers.google.com/drive/api/v3/reference/files/update
- Reddit Rest API: https://www.reddit.com/dev/api#POST_api_v1_collections_add_post_to_collection

I transformed used a query parameter in this case to achieve a clear separation between resource representation and additional fields that don't belong to the resource itself.

> I think we should simply process the `published` boolean field from the body in both `create`(POST) and `update`(PATCH), according to our resource representation: https://github.com/camunda/camunda-docs-cawemo/blob/8794-public-api/content/reference/rest-api/templates/_index.md#resource-representation.

I initially thought about that, too, but decided against it doesn't sound as descriptive for what is happening as `publishVersion`.

To me it seems like there is a need for more in-depth discussion about the design principles of the rest api. I'll schedule another meeting with you and Volker to clarify some points so we can finalize this.

Reply:
> > I don't think sending query parameters in `POST` or `PATCH` is a good practice.
> 
> Sending query parameters in `POST` and `PATCH` requests is a widely used practice across the industry. To give you some examples:
> 
> * Twitter Rest API: https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/mute-block-report-users/api-reference/post-blocks-create
> * Google Drive Rest API: https://developers.google.com/drive/api/v3/reference/files/update
> * Reddit Rest API: https://www.reddit.com/dev/api#POST_api_v1_collections_add_post_to_collection
> * ElasticSearch Rest API: https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html
> 
> I transformed used a query parameter in this case to achieve a clear separation between resource representation and additional fields that don't belong to the resource itself.
> 
> >

I get your point, it makes sense in some situations.
I had a look at a couple of examples you gave and in general, the query parameters are used for instructions necessary for the creation/update of the entity, not for providing a value to a field of the resource. (maybe it doesn't apply to all your examples).
> I transformed used a query parameter in this case to achieve a clear separation between resource representation and additional fields that don't belong to the resource itself.

Regardless of that, the resource representation that you share with the customer: https://github.com/camunda/camunda-docs-cawemo/blob/8794-public-api/content/reference/rest-api/templates/_index.md#resource-representation is showing `published` as being a field of the resource.
That's why I see the query parameter as being redundant.

Reply:
During our follow-up meeting (see https://docs.google.com/document/d/1hHLxyPrqvqORjocnoYJDBl22-_wpeMBR-iyx6YYuxbw/edit?usp=sharing) we agreed on aligning the resource representations of the modeler api and the public api (see https://github.com/camunda/camunda-docs-cawemo/pull/118#discussion_r692853390) with the option to add additional information into the `metadata` field.

We decided to use this instead of a query parameter as its cleaner and more extensible in case of future iterations of the template endpoints of the public api.